integration-tests/src/test/java/io/druid/tests/indexer/ITRealtimeIndexTaskTest.java

```diff
@@ -132,10 +213,57 @@ public void postEvents() throws Exception
           httpClient,
           smileMapper
       );
-      client.postEventsFromFile(EVENT_DATA_FILE);
+      // there are 22 lines in the file
+      int i = 1;
+      DateTime dt = new DateTime(zone);  // timestamp used for sending each event
+      dtFirst = dt;                      // timestamp of 1st event
+      dtLast = dt;                       // timestamp of last event
+      String line;
+      while ((line = reader.readLine()) != null) {
+        if (i == 15) { // for the 15th line, use a time before the window
+          dt = dt.minusMinutes(10);
+        } else if (i == 18) { // use a time 6 seconds ago so it will be out of order
+          dt = dt.minusSeconds(6);
+        }
+        String event = line.replace(TIME_PLACEHOLDER, EVENT_FMT.print(dt));
+        LOG.info("sending event: \n" + event);
```

Comment:
can you use formatter instead of string concatenation? (here and in all other places if applicable)